src/memory_map.rs

```diff
@@ -67,33 +68,32 @@ impl MemoryArea {
 
     /// The type of the memory region.
     pub fn typ(&self) -> MemoryAreaType {
-        match self.typ {
-            1 => MemoryAreaType::Available,
-            3 => MemoryAreaType::AcpiAvailable,
-            4 => MemoryAreaType::ReservedHibernate,
-            5 => MemoryAreaType::Defective,
-            _ => MemoryAreaType::Reserved,
-        }
+        self.typ
     }
 }
 
 /// An enum of possible reported region types.
-#[derive(Debug, PartialEq, Eq)]
+/// Inside the Multiboot2 spec this is kind of hidden
+/// inside the implementation of `struct multiboot_mmap_entry`.
+#[derive(Debug, PartialEq, Eq, Copy, Clone)]
+#[repr(u32)]
 pub enum MemoryAreaType {
-    /// A reserved area that must not be used.
-    Reserved,
-
     /// Available memory free to be used by the OS.
-    Available,
+    Available = 1,
+
+    /// A reserved area that must not be used.
+    Reserved = 2,
 
     /// Usable memory holding ACPI information.
-    AcpiAvailable,
+    AcpiAvailable = 3,
 
     /// Reserved memory which needs to be preserved on hibernation.
-    ReservedHibernate,
+    /// Also called NVS in spec, which stands for "Non-Volatile Sleep/Storage",
```

Comment:
Good to mention it's called `NVS` in the C code, but I think we should preserve the bit about this memory needing to be preserved when hibernating.